lms/djangoapps/certificates/tests/test_signals.py

```diff
@@ -184,3 +185,50 @@ def test_cert_generation_on_passing_instructor_paced(self):
                         student=self.user,
                         course_key=self.ip_course.id
                     )
+
+
+class LearnerTrackChangeCertsTest(ModuleStoreTestCase):
+    """
+    Tests for certificate generation task firing on learner verification
+    """
+    def setUp(self):
+        super(LearnerTrackChangeCertsTest, self).setUp()
+        self.course = CourseFactory.create(self_paced=True)
+        self.user = UserFactory.create()
+        self.enrollment = CourseEnrollmentFactory(
+            user=self.user,
+            course_id=self.course.id,
+            is_active=True,
+            mode="verified",
+        )
+        self.ip_course = CourseFactory.create(self_paced=False)
+
+    def test_cert_generation_on_photo_verification_self_paced(self):
+        with mock.patch(
+            'lms.djangoapps.certificates.signals.generate_certificate.apply_async',
+            return_value=None
+        ) as mock_generate_certificate_apply_async:
+            with waffle.waffle().override(waffle.SELF_PACED_ONLY, active=True):
+                with mock_passing_grade():
+                    attempt = SoftwareSecurePhotoVerification(user=self.user)
+                    attempt.status = "approved"
+                    attempt.save()
+                    mock_generate_certificate_apply_async.assert_called(
```

Comment:
On the other hand, these tests _do_ seem to imply that the certificate is getting generated...